src/main/java/anto/Storage.java

```diff
@@ -0,0 +1,214 @@
+    private Path absolutePath;
+    private File antoFile;
+
+    /**
+     * Creates a Storage class.
+     *
+     * @param ui Ui that handles all printing to the command line.
+     * @param filePath Relative file path of file to load data from.
+     */
+    public Storage(Ui ui, String filePath) {
+        this.ui = ui;
+        this.relativePath = filePath;
+        this.absolutePath = Paths.get(relativePath).toAbsolutePath();
+        this.antoFile = absolutePath.toFile();
+
+    }
+
+    /**
+     * Checks file at file path and loads data if it is there, else create a new file.
+     *
+     * @return ArrayList of Tasks representing data.
+     * @throws AntoException Throws exception if there is an IO Exception.
+     */
+    public ArrayList<Task> loadSave() throws AntoException {
+        try {
+            ArrayList<Task> taskList = new ArrayList<>();
+
+            // If file doesn't exist
+            if (!antoFile.exists()) {
+                antoFile.createNewFile();
+                this.ui.printNoSavedFile();
+            } else {
+                Scanner sc = new Scanner(antoFile);
+                while (sc.hasNextLine()) {
+                    String currLine = sc.nextLine();
+                    String[] currLineArr = currLine.split(Pattern.quote(" | "));
+                    if (currLineArr[0].equals("T")) {
+                        Task newTask = new Todo(currLineArr[2]);
+                        if (currLineArr[1].equals("1")) {
+                            newTask.markAsDone();
+                        }
+                        taskList.add(newTask);
+                    } else if (currLineArr[0].equals("D")) {
+                        Task newTask = new Deadline(currLineArr[2], currLineArr[3]);
+                        if (currLineArr[1].equals("1")) {
+                            newTask.markAsDone();
+                        }
+                        taskList.add(newTask);
+                    } else if (currLineArr[0].equals("E")) {
+                        Task newTask = new Event(currLineArr[2], currLineArr[3], currLineArr[4]);
+                        if (currLineArr[1].equals("1")) {
+                            newTask.markAsDone();
+                        }
+                        taskList.add(newTask);
+                    }
+                }
+                this.ui.printSavedFileFound(taskList);
+            }
+            return taskList;
+        } catch (java.io.IOException e) {
+            throw new AntoException("OOPS!!! IOException");
+        }
+    }
+
+    /**
+     * Adds task to Anto file.
+     *
+     * @param task Task to be added.
+     * @throws AntoException Throws AntoException when there is an IOException.
+     */
+    public void addTaskToStorage(Task task) throws AntoException {
+        try {
+            FileWriter writer = new FileWriter(antoFile, true);
+            if (task instanceof Todo) {
+                // It is safe to type cast because the type is checked before
```

Comment:
I like these comments that highlight typecasting safety. Although maybe they could be condensed into 1 comment at the top?